hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -2430,6 +2432,71 @@ private class Emit(
           }
         }
 
+      case x@NDArraySlice(child, slicesIR) =>
+        val childEmitter = deforest(child)
+
+        val slicest = emit(slicesIR, env, resultRegion, None)
+        val slicesValueAddress = mb.newField[Long]
+        val slicesTuple = new CodePTuple(coerce[PTuple](slicesIR.pType), region, slicesValueAddress)
+
+        val codeSlices = ArrayBuffer[(Code[Long], Code[Long], Code[Long])]()
+        val codeRefMap = mutable.Map[Int, Code[Long]]()
+
+        var missingSliceElements = const(false)
+
+        coerce[PTuple](slicesIR.pType).types.zipWithIndex.foreach { case (sliceOrIndex, dim) =>
+          sliceOrIndex match {
+            case p: PTuple => {
+              val oneSlice = new CodePTuple(p, region, slicesTuple(dim))
+              missingSliceElements = missingSliceElements || oneSlice.isMissing(0) || oneSlice.isMissing(1) || oneSlice.isMissing(2)
+              codeSlices += ((oneSlice[Long](0), oneSlice[Long](1), oneSlice[Long](2)))
+            }
+            case _: PInt64 => {
+              codeRefMap += dim -> slicesTuple(dim)
+            }
+          }
+        }
+
+        val outputShape = codeSlices.toArray.map { case (start, stop, step) =>
```

Comment:
Any reason to have the toArray before the map? I think it reads more clearly to coerce to the type of interest as the last step.